youtube_comedy_slam/youtube_comments_api.py

```diff
@@ -0,0 +1,153 @@
+            first = True
+            for item in results["items"]:
+                if first:
+                    first = False
+                    continue
+                snippet = item["snippet"]
+                comment = snippet["topLevelComment"]["snippet"]
+                videoId = comment["videoId"]
+                textDisplay = comment["textDisplay"]
+                canReply = snippet["canReply"]
+                totalReplyCount = snippet["totalReplyCount"]
+                kind = item["kind"]
+                etag = snippet["topLevelComment"]["etag"]
+                id = snippet["topLevelComment"]["id"]
+                authorChannelId = comment["authorChannelId"] if "authorChannelId" in comment else ""
+                authorDisplayName = comment["authorDisplayName"]
+                likeCount = comment["likeCount"]
+                publishedAt = comment["publishedAt"]
+                updatedAt = comment["updatedAt"]
+                viewerRating = comment["viewerRating"]
+                writer.writerow(
+                    {'viewerRating': viewerRating, 'videoId': videoId, 'textDisplay': textDisplay, 'canReply': canReply,
+                     'totalReplyCount': totalReplyCount, 'kind': kind, 'etag': etag, 'id': id,
+                     'authorChannelId': authorChannelId, 'authorDisplayName': authorDisplayName, 'likeCount': likeCount,
+                     'publishedAt': publishedAt, 'updatedAt': updatedAt, 'viewerRating': viewerRating, 'videoId': videoId})
+
+        get_comment_page(results)
+
+        counter = 0
+        while ('nextPageToken' in results) and (counter < 3):
+            # Fetch next page
+            nextPageToken = results['nextPageToken']
+            results = youtube.commentThreads().list(
+                part="snippet,replies",
+                textFormat="plainText",
+                maxResults=100,
+                pageToken = nextPageToken,
+                videoId=video_id
+            ).execute()
+            # You have to ignore the first result
+            get_comment_page(results, ignoreFirst=True)
+            counter += 1
+
+    return results
+
+
+def get_comments(youtube, parent_id):
+    results = youtube.comments().list(
+        part="snippet",
+        parentId=parent_id,
+        textFormat="plainText"
+    ).execute()
+
+    print(results)
+
+    for item in results["items"]:
+        author = item["snippet"]["authorDisplayName"]
+        text = item["snippet"]["textDisplay"]
+        print("Comment by %s: %s" % (author, text))
+
+    return results["items"]
+
+
+def comment_threads_list_all_threads_by_channel_id(service, part, all_threads_related_to_channel_id):
+    results = service.commentThreads().list(
+        allThreadsRelatedToChannelId=all_threads_related_to_channel_id,
+        part=part
+    ).execute()
+
+    print_results(results)
+
+
+# comment_threads_list_all_threads_by_channel_id(youtube, 'snippet,replies', '7zCIRPQ8qWc')
```

Comment:
You should run the code from the block starting by:
`if __name__ == '__main__':`

Also use argparser for quick experimenting.